src/GafferOSL/OSLShader.cpp

```diff
@@ -604,13 +600,13 @@ Plug *loadMatrixArrayParameter( const OSLQuery::Parameter *parameter, const Inte
 
 Plug *loadClosureParameter( const OSLQuery::Parameter *parameter, const InternedString &name, Gaffer::Plug *parent )
 {
-	Plug *existingPlug = parent->getChild<Plug>( name );
+	ClosurePlug *existingPlug = parent->getChild<ClosurePlug>( name );
 	if(	existingPlug && existingPlug->typeId() == Plug::staticTypeId() )
```

Comment:
This test is going to fail now, and we'll end up replacing plugs that were perfectly good. I think now it just needs to be the following :

```
if( existingPlug )
{
    return existingPlug;
}
```